### PR TITLE
Add transaction example

### PR DESCRIPTION
**Pull request by edenhill**

### CODE REVIEW

**Comment by nick-zh** [comment]
thanks very much!

File: INTRODUCTION.md [hunk 1]
```diff
@@ -723,65 +723,27 @@ This method should be called by the application on delivery report error.
...
    must call `rd_kafka_abort_transaction()`, rewind its input to the
    point before the current transaction started, and attempt a new transaction
    by calling `rd_kafka_begin_transaction()`, etc.
+   Use `rd_kafka_error_is_txn_abortable()` to distinguish this case.
+ * Fatal errors - the application must cease operations and destroy the
+   producer instance.
+   Use `rd_kafka_error_is_fatal()` to distinguish this case.
+ * For all other errors returned from the transactional API: the current
+   recommendation is to treat any error that has neither retriable, abortable,
+   or fatal set, as a fatal error.
```
Comment (mhowlett) [inline INTRODUCTION.md]: this all reads well to me. fwiw, what i'm wondering about having read this is what to do with errors that occurred whilst using the transactional producer from methods that aren't part of the transactional api.

File: src-cpp/rdkafkacpp.h [hunk 2]
```diff
@@ -909,7 +909,7 @@ class RD_EXPORT RebalanceCb {
    *           consumer->unassign();
    *
    *         } else {
-   *           std::cerr << "Rebalancing error: <<
+   *           std::cerr << "Rebalancing error: " <<
```
Comment (mhowlett) [inline src-cpp/rdkafkacpp.h]: !

File: examples/transactions.c [hunk 3]
```diff
@@ -0,0 +1,656 @@
...
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @name Transactions example
+ *
+ * This example show-cases a simple transactional consume-process-produce
+ * application.
```
Comment (mhowlett) [inline examples/transactions.c]: maybe a one or two sentence description of more specifically what it does.

File: examples/transactions.c [hunk 4]
```diff
@@ -0,0 +1,656 @@
...
+
+/**
+ * @brief Message delivery report callback.
+ *
+ * This callback is called exactly once per message, indicating if
+ * the message was succesfully delivered
+ * (rkmessage->err == RD_KAFKA_RESP_ERR_NO_ERROR) or permanently
+ * failed delivery (rkmessage->err != RD_KAFKA_RESP_ERR_NO_ERROR).
+ *
+ * The callback is triggered from rd_kafka_poll(), rd_kafka_flush(),
```
Comment (mhowlett) [inline examples/transactions.c]: 'is triggered as a side effect of'?

**Pull request merged**